packages/base-driver/lib/protocol/protocol.js

```diff
@@ -283,12 +291,15 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
         `${driver.constructor.name}.${spec.command}() with args: ` +
         _.truncate(JSON.stringify(args), {length: MAX_LOG_BODY_LENGTH}));
 
-      if (driver.executeCommand) {
-        driverRes = await driver.executeCommand(spec.command, ...args);
-      } else {
-        driverRes = await driver.execute(spec.command, ...args);
+      if (pluginOverrodeProxy) {
+        // TODO for now we add this information on the args list, but that's mixing purposes here.
+        // We really should add another 'options' parameter to 'executeCommand', but this would be
+        // a breaking change for all drivers so would need to be handled carefully.
+        args.push({reqForProxy: req});
       }
 
+      driverRes = await driver.executeCommand(spec.command, ...args);
```

Comment:
What is `driver.execute()`?

Reply:
it was an old way of doing what `executeCommand` does that we left for backwards compatibility with older appium drivers. this probably could have been removed years ago and just doing it now since i read the TODO.